### DISCUSSION TITLE
Upgraded to v2: process is undefined

### DISCUSSION THREAD

**Discussion by ap1969** [quasarframework/quasar#10088] [CLI - PWA mode] [answered]
Hi team,

I've just updated an app to v2, and it's working great (xell done!) apart from one section:

```
if (process.env.DEV) {
        return process.env.API_URL
      } else {
        return ""
      }
```

According to https://quasar.dev/quasar-cli/handling-process-env , this should work. Any idea why not? (It used to work in v1).

The error is: vue-router.esm-bundler.js?f6c4:3248 ReferenceError: process is not defined

Regards,
Andy

**Comment by justinbeaty** [comment, marked as answer]
@ap1969 check this question to see if it helps: #9967

**Reply by ap1969** [reply]
Thanks @justinbeaty  - it kind of helped.

Originally, with v1, I was doing:

quasar.conf.js:

`const envVars = require('dotenv').config({ path: `.env.${process.env.NODE_ENV}` }).parsed;`

then setting build.env = envVars.

This worked fine.

now it doesn't.

If I change it to explicity say: 

```
build.env  = {
    X: "y",
    key: "val"
 }
```

it works. Bit of a pain, as I then need to update the envvars in two places. I'll keep looking into it and see why one works and not the other, as that makes no sense.

Andy

**Reply by ap1969** [reply]
Hmmm. That's really wierd, I just debugging what dotenv was picking up, and it's the same as I'd updated the build.env config setting to. 

So I reset it back to build.env = envVars , and it worked fine.

Very strange.

**Reply by justinbeaty** [reply]
That's very strange, because yeah it should have worked normally with `build.env: envVars`

Either way, glad it's working, maybe you just needed to restart `quasar dev`

**Comment by ap1969** [comment]
I think I got to the bottom of this.

It looks like if the env vars are defined as 

X: "123",
Y: "456",

and I try to call process.env.Z , then the error "process is not defined" is generated (not "process.env is not defined", as you'd expect).

**Reply by calvinf** [reply]
💯 Super helpful comment, same thing for me, my env file was missing something that should have been there and causing a "process.env is not defined" error when what was really happening is that the variable was missing from process.env because it wasn't in the `.env`.
